src/Abstractions/TaskOptions.cs

```diff
@@ -6,153 +6,61 @@ namespace Microsoft.DurableTask;
-        /// </summary>
-        /// <param name="handler">The handler to invoke when deciding whether to retry a failed orchestrator task.</param>
-        /// <returns>Returns the current <see cref="Builder"/> object.</returns>
-        public Builder WithRetryStrategy(AsyncRetryHandler handler)
-        {
-            this.RetryHandler = handler;
-            this.RetryPolicy = null;
-            return this;
-        }
-
-        /// <summary>
-        /// Configures a <see cref="CancellationToken"/> that can be used to cancel the task execution.
-        /// </summary>
-        /// <remarks>
-        /// Cancellation tokens can be used to stop the current orchestrator from awaiting a pending activity or
-        /// sub-orchestration completion. However, this cancellation won't necessarily stop the activity or
-        /// sub-orchestration from running in the background.
-        /// </remarks>
-        /// <param name="cancellationToken">The cancellation token to use for cancelling task execution.</param>
-        /// <returns>Returns the current <see cref="Builder"/> object.</returns>
-        public Builder WithCancellationToken(CancellationToken cancellationToken)
-        {
-            if (cancellationToken != default)
-            {
-                throw new NotSupportedException(
-                    "Durable task cancellation is not yet supported. See"
-                    + " https://github.com/microsoft/durabletask-dotnet/issues/7 for more information.");
-            }
-
-            this.CancellationToken = cancellationToken;
-            return this;
-        }
+/// <summary>
+/// Options that can be used to control the behavior of orchestrator task execution. This derived type can be used to
+/// supply extra options for orchestrations.
+/// </summary>
+/// <param name="InstanceId">The orchestration instance ID to use. <c>null</c> to have one generated.</param>
+/// <param name="Retry">The retry options. <c>null</c> for no retries.</param>
+public record OrchestrationOptions(string? InstanceId = null, TaskRetryOptions? Retry = null)
+    : TaskOptions(Retry);
 
-        /// <summary>
-        /// Creates a new <see cref="TaskOptions"/> object from this builder.
-        /// </summary>
-        /// <returns>The created <see cref="TaskOptions"/> object.</returns>
-        public TaskOptions Build() => new(this);
-    }
-}
+/// <summary>
+/// Options for submitting new orchestrations via the client.
+/// </summary>
+/// <param name="InstanceId">
+/// The unique ID of the orchestration instance to schedule. If not specified, a new GUID value is used.
+/// </param>
+/// <param name="StartAt">
+/// The time when the orchestration instance should start executing. If not specified or if a date-time in the past
+/// is specified, the orchestration instance will be scheduled immediately.
+/// </param>
+public record StartOrchestrationOptions(string? InstanceId = null, DateTimeOffset? StartAt = null);
```

Comment:
Similar here re: naming. Alternative could be `NewOrchestrationOptions`.

Reply:
The risky thing about `NewOrchestrationOptions` is that users might wrongly assume that it also applies to sub-orchestrations.